src/main/java/sorald/event/models/miner/MinedRule.java

```diff
@@ -0,0 +1,29 @@
+package sorald.event.models.miner;
+
+import com.google.common.collect.ImmutableList;
+import sorald.event.models.WarningLocation;
+
+public class MinedRule {
+    private final String ruleKey;
+    private final String ruleName;
+    private final ImmutableList<WarningLocation> warningLocations;
+
+    public MinedRule(
+            String ruleKey, String ruleName, ImmutableList<WarningLocation> warningLocations) {
+        this.ruleKey = ruleKey;
+        this.ruleName = ruleName;
+        this.warningLocations = warningLocations;
+    }
+
+    public ImmutableList<WarningLocation> getWarningLocations() {
+        return warningLocations;
+    }
```

Comment:
Alternatively, one can have the internal list be mutable, and then present an immutable copy of it.

```suggestion
   /** @returns An immutable list of warning locations. */
    public List<WarningLocation> getWarningLocations() {
        return Collections.unmodifiableList(warningLocations);
    }
```